Add automated builds

Flynsarmy opened the pull request:
This PR adds a git workflow that automatically builds for every platform on push or pull_request. Having automated builds makes testing PRs or specific commits a lot easier for people wanting to contribute to the project.

The build creates an MTerrain.zip artifact:
![image](https://github.com/user-attachments/assets/09af37cd-4ee0-4399-96d1-f6fa5a915c62)
that contains a ready to go addons folder with binaries for each of the platforms below:
![image](https://github.com/user-attachments/assets/83885c3a-8920-47e4-90cf-f054fbef9958)

Notes: 
- I couldn't get linux x86_32 compiling. Not sure why. The binary names are slightly different to what you had in your manual compiles of this addon and I updated MTerrain.gdextension accordingly.
- I had to update the godot-cpp commit to get this thing compiling. I set it to the 4.3 branch but we can always change it to 4.1 to get compatibility with all recent versions of godot.

mohsenph69 commented:
Hi there, I am not sure how this works! but you have build for **double precision** and also changed  The binary names to double version! I think by default it should **single precision** as most people use single one! or if you know a way which can automatically change **libs/MTerrain.gdextension** file base on single or double precision would be good!

Flynsarmy commented:
> I am not sure how this works! 

When a push or pull request is made to this repo, the following workflow is kicked off:
![image](https://github.com/user-attachments/assets/9c5d41ce-df6b-441f-8209-51a195d8ab62)
Each one of those build jobs on the left will create a binary - one for windows.x86_64.debug, one for linux.x86_32.release etc. The binaries are all saved to github artifacts which essentially is just a storage bucket.

If all of the builds succeed, the artifacts are extracted and placed into a copy of the libs folder of this repo. The whole thing is zipped up and saved as another artifact that the end user can then download:
![image](https://github.com/user-attachments/assets/988a24f5-58de-4b49-8e95-2eb09585b4d7)


> but you have build for **double precision** and also changed The binary names to double version! I think by default it should **single precision** as most people use single one! 

Done!

> or if you know a way which can automatically change **libs/MTerrain.gdextension** file base on single or double precision would be good!

Now that we're back on single precision I reverted the filenames back to how you originally had them. Dynamically updating *MTerrain.gdextension* would be fairly easy to do with a bash script inside *.github/workflows/group_builds.yml* though as I said all automated builds are back to single precision now so not really necessary at the moment.

nickpolet commented:
Would be nice to see this added. Going to put a vote in there for including double precision builds as a workflow/action. Would be nice to download single or double versions if possible.
